libbeat/processors/add_process_metadata/add_process_metadata.go

```diff
@@ -105,7 +106,7 @@ func newProcessMetadataProcessorWithProvider(cfg *common.Config, provider proces
 	p := addProcessMetadata{
 		config:      config,
 		provider:    provider,
-		cidProvider: cidProvider,
+		cidProvider: newCidProvider(config.HostPath, config.CgroupPrefixes, processCgroupPaths),
```

Comment:
I recommend leaving this as nil when `container.id` is not in the `mappings`. Then adjust the enrich function accordingly. This will save it from doing the work when it's disabled.

Reply:
Thanks for the suggestion!
Instead of making the `cidProvider` nil when it is disabled, I passed a dummy function to the `cidProvider` which also avoids the use of `cgroup.ProcessCgroupPaths` to see the `cgroup` of the process.
The reason for that is I felt that the use of nil is a code smell, as we could not extend nil in the future. Also, the extra steps introduced by the dummy function will probably be optimized away by the compiler.